src/main/java/seedu/moneygowhere/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +35,51 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DATE, PREFIX_COST, PREFIX_REMARK, PREFIX_TAG);
+
+        List<Predicate<Spending>> predicates = new ArrayList<>();
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            String name = argMultimap.getValue(PREFIX_NAME).get();
+            String[] nameKeywords = name.split("\\s+");
+            predicates.add(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        }
+        if (argMultimap.getValue(PREFIX_DATE).isPresent()) {
+            List<Date> dates = ParserUtil.parseDates(argMultimap.getAllValues(PREFIX_DATE));
+
+            if (dates.size() < 2) {
+                throw new ParseException(Date.MESSAGE_CONSTRAINTS);
+            }
+            predicates.add(new DateInRangePredicate(dates.get(0), dates.get(1)));
         }
+        if (argMultimap.getValue(PREFIX_COST).isPresent()) {
+            List<Cost> costs = ParserUtil.parseCosts(argMultimap.getAllValues(PREFIX_COST));
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+            double min = Double.parseDouble(costs.get(0).value);
+            double max = Double.parseDouble(costs.get(1).value);
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
-    }
+            if (costs.size() != 2 || min > max) {
+                throw new ParseException(Cost.MESSAGE_CONSTRAINTS);
+            }
+
+            predicates.add(new CostInRangePredicate(costs.get(0), costs.get(1)));
+        }
+        if (argMultimap.getValue(PREFIX_REMARK).isPresent()) {
+            String name = argMultimap.getValue(PREFIX_REMARK).get();
+            String[] nameKeywords = name.split("\\s+");
+            predicates.add(new RemarkContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        }
+        if (argMultimap.getValue(PREFIX_TAG).isPresent()) {
```

Comment:
Are we allowing > 1 tag e.g. `find t/food t/transport`?

Reply:
Currently only implemented for one tag, but yes, this is a consideration.

Reply:
Implemented: `find t/food transport`

No need to indicate `t/` again